sbin/pfdhcplistener

```diff
@@ -244,6 +249,54 @@ sub process_pkt {
 
             # we send all IPv4 DHCPv4 codepath
             if ( $l2->{type} eq ETH_TYPE_IP ) {
+                my ($dhcp);
+
+                # we need success flag here because we can't next inside try catch
+                my $success;
+                try {
+                    $dhcp = decode_dhcp($l4->{'data'});
+                    $success = 1;
+                } catch {
+                    $logger->warn("Unable to parse DHCP packet: $_");
+                };
+                return if (!$success);
+                $args{dhcp} = $dhcp;
+                
+                # adding to dhcp hashref some frame information we care about
+                $dhcp->{'src_mac'} = $args{'src_mac'};
+                $dhcp->{'dest_mac'} = $args{'dest_mac'};
+                $dhcp->{'src_ip'} = $args{'src_ip'};
+                $dhcp->{'dest_ip'} = $args{'dest_ip'};
+
+
+                # grab DHCP information
+                if ( !defined($dhcp->{'chaddr'}) ) {
+                    $logger->debug("chaddr is undefined in DHCP packet");
+                    return;
+                }
+
+                $dhcp->{'chaddr'} = clean_mac( substr( $dhcp->{'chaddr'}, 0, 12 ) );
+
+                if (!valid_mac($dhcp->{'src_mac'})) {
+                    $logger->debug("Source MAC is invalid. skipping");
+                    return;
+                }
+
+                if ( $dhcp->{'chaddr'} ne "00:00:00:00:00:00" && !valid_mac($dhcp->{'chaddr'}) ) {
+                    $logger->debug( sub {
+                        "invalid CHADDR value ($dhcp->{'chaddr'}) in DHCP packet from $dhcp->{src_mac} ($dhcp->{src_ip})"
+                    });
+                    return;
+                }
+
+
+                my $dhcp_mac = $dhcp->{'chaddr'};
+                # If its a DHCPREQUEST, we take the IP address from the options
+                my $dhcp_ip = ( $dhcp->{'options'}{'53'} == 3 ) ? $dhcp->{'options'}{'50'} : $dhcp->{yiaddr};
```

Comment:
It seams that sometimes $dhcp_ip is empty, we probably have to add a condition if define($dhcp_ip) ...after in the code